services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,62 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+  if (event == 0){
+    return ERR_CODE_INVALID_ARG;
+  }
+
   /* Send an event to the thermal manager queue */
+  if (xQueueSend(thermalMgrQueueHandle, event, (TickType_t)5) != pdPASS){
+    return ERR_CODE_UNKNOWN; // failed to send message
+  }
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event = {};
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&event); //OS interrupt occurred, send OS interrupt event
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+
+  thermal_mgr_event_t buffer = {};
+
+  float tempC = 0;
+
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &buffer, (TickType_t) 5) != pdPASS){ 
```

Comment:
Use should be using portMAX_DELAY instead of 5 here as we want to block the task until an event is received in the queue